### PR TITLE
Update version.info to `2.5.0.dev`

### PR DESCRIPTION
**Pull request by rasbt**
This PR increases the version since there have been a bunch of changes/fixes since the last release. This makes it a bit easier to detect which version is currently installed in a system.

Ideally, could we also make a new release (or release candidate) soon @Borda ?

----
📚 Documentation preview 📚: https://pytorch-lightning--20316.org.readthedocs.build/en/20316/

### CODE REVIEW

**Comment by t-vi** [comment]
Can we use the PyTorch scheme of `2.5.0a0` though? This will sort before the final 2.5.0 in pip (and other Python standard) versioning.

**Comment by Borda** [comment]
> Can we use the PyTorch scheme of `2.5.0a0` though? This will sort before the final 2.5.0 in pip (and other Python standard) versioning.

I don't think that we can do 2.5 any time some as we didn't landed any feature since 2.4

**Comment by t-vi** [comment]
The version format should still use FOOa0 instead of FOO.dev . There is a pep for it and then the versions compare properly.

In terms of features, I would love for litgpt to allow newer bitsandbytes with Ali's recent PR, in 2.4.1 or so.


Am 18. Oktober 2024 21:41:34 MESZ schrieb Jirka Borovec ***@***.***>:

**Comment by rasbt** [comment]
@t-vi  Do you mean [PEP 440](https://peps.python.org/pep-0440/)? That's a good point, but I think it may not matter anymore with more modern packaging tools. E.g., it looks like the sorting order is correct (the development version coming before the alpha release etc):


```python
from packaging.version import Version

versions = [
    "1.0.0",
    "1.0.1",
    "1.0.0a0",  # Alpha 0 version
    "1.0.0a1",  # Alpha 1 version
    "1.0.0b1",  # Beta version
    "1.0.0rc1", # Release candidate
    "1.0.0.post1",  # Post-release
    "1.0.0.dev1",  # Development version
    "1.0.0.post2",  # Another post-release
]

version_objs = [Version(v) for v in versions]

sorted_versions = sorted(version_objs)

for version in sorted_versions:
    print(version)
```

prints

```
1.0.0.dev1
1.0.0a0
1.0.0a1
1.0.0b1
1.0.0rc1
1.0.0
1.0.0.post1
1.0.0.post2
1.0.1
```

Personally, I find it a bit more natural to have the Gh main branch version named `.dev1` (and then switching to `a0` when creating a pre-release tag), but I don't have a strong preference. 

I totally agree with you though that we should have a pre-release (even  if it's just an alpha) out soon so that we can finally update bitsandbytes in LitGPT.

**Comment by rasbt** [comment]
@Borda 

> I don't think that we can do 2.5 any time some as we didn't landed any feature since 2.4

Can we do a 2.4.1 please though?

**Comment by Borda** [comment]
> > I don't think that we can do 2.5 any time some as we didn't landed any feature since 2.4
> 
> Can we do a 2.4.1 please though?

Sure but that goes from `release/stable` branch not master...

**Comment by rasbt** [comment]
> Sure but that goes from release/stable branch not master...

Yes, that's a different discussion. For now, could we have the version in master updated to .dev1 or a0 though so that we can can more easily find out check which version a user is on?

**Review by Borda — APPROVED** [review]

**Comment by speediedan** [comment]
Is there any update on whether a Lightning `2.5` release is planned? My sense is that this release may be skipped but I wanted to ensure that was the case before I issued a [Fine-Tuning Scheduler](https://finetuning-scheduler.readthedocs.io/en/latest/) pre-release for `2.5` and focused work on PyTorch `2.6` trunk. Thank you all again for all your contributions!
@Borda @t-vi @lantiga @rasbt

**Comment by lantiga** [comment]
Hey @speediedan, 2.5 is totally coming :) expected in the next 2 weeks
Let us know if you need a specific PR to go in

**Comment by speediedan** [comment]
> Hey @speediedan, 2.5 is totally coming :) expected in the next 2 weeks Let us know if you need a specific PR to go in

Awesome! No specific PR needed at the moment, just needed to know for release planning purposes. I was going to issue a pre-release for users that wanted to use Fine-Tuning Scheduler's experimental [`Auto` FSDP2 plan](https://finetuning-scheduler.readthedocs.io/en/latest/distributed/model_parallel_scheduled_fine_tuning.html#fts-auto-fsdp2-plan-configuration) functionality if no Lightning `2.5` release was planned. Thanks Luca!
